internals/plan/plan.go

```diff
@@ -765,6 +771,12 @@ func CombineLayers(layers ...*Layer) (*Layer, error) {
 					Message: fmt.Sprintf("plan check %q command invalid: %v", name, err),
 				}
 			}
+			if check.Exec.Context != "" && !serviceNames[check.Exec.Context] {
+				return nil, &FormatError{
+					Message: fmt.Sprintf("plan check %q context %q is not a service name",
+						name, check.Exec.Context),
```

Comment:
Not a huge fan of the error message here. IMO it's not `%q is not a service name` but more like `service %q not found`. Maybe
`plan check %q context: service %q not found`
or
`plan check %q context specifies non-existent service %q`